### ISSUE TITLE
About warping result mask

### ISSUE THREAD

**Issue by ShaofengZou** [centreborelli/mf2f#3]
In ```compute_warping_error.sh```, the definition of plambda, blur, downsa, upsa is missing. 
![image](https://user-images.githubusercontent.com/15974452/109798161-6005f680-7c55-11eb-8692-688ab3fb440d.png)
Could you give any hint about this?

**Comment by oooo1114** [comment]
I got the same error and an additional path error. Please help, thanks a lot.
```
compute_warping_error.sh: line 11: /home/dewil/occlu-mask/build/bin/warp-bicubic: No such file or directory
compute_warping_error.sh: line 14: plambda: command not found
compute_warping_error.sh: line 19: downsa: command not found
compute_warping_error.sh: line 20: downsa: command not found
compute_warping_error.sh: line 23: plambda: command not found
compute_warping_error.sh: line 23: downsa: command not found
compute_warping_error.sh: line 26: blur: command not found
compute_warping_error.sh: line 27: blur: command not found
compute_warping_error.sh: line 29: blur: command not found
compute_warping_error.sh: line 29: plambda: command not found
compute_warping_error.sh: line 33: plambda: command not found
compute_warping_error.sh: line 38: blur: command not found
compute_warping_error.sh: line 40: blur: command not found
compute_warping_error.sh: line 40: plambda: command not found
compute_warping_error.sh: line 44: upsa: command not found
compute_warping_error.sh: line 45: upsa: command not found
compute_warping_error.sh: line 45: plambda: command not found
```

**Comment by oooo1114** [comment]
> I got the same error and an additional path error. Please help, thanks a lot.
> 
> ```
> compute_warping_error.sh: line 11: /home/dewil/occlu-mask/build/bin/warp-bicubic: No such file or directory
> compute_warping_error.sh: line 14: plambda: command not found
> compute_warping_error.sh: line 19: downsa: command not found
> compute_warping_error.sh: line 20: downsa: command not found
> compute_warping_error.sh: line 23: plambda: command not found
> compute_warping_error.sh: line 23: downsa: command not found
> compute_warping_error.sh: line 26: blur: command not found
> compute_warping_error.sh: line 27: blur: command not found
> compute_warping_error.sh: line 29: blur: command not found
> compute_warping_error.sh: line 29: plambda: command not found
> compute_warping_error.sh: line 33: plambda: command not found
> compute_warping_error.sh: line 38: blur: command not found
> compute_warping_error.sh: line 40: blur: command not found
> compute_warping_error.sh: line 40: plambda: command not found
> compute_warping_error.sh: line 44: upsa: command not found
> compute_warping_error.sh: line 45: upsa: command not found
> compute_warping_error.sh: line 45: plambda: command not found
> ```

solve the 1st one by change line11 in 'compute_warping_error.sh' from ' WARP3="/home/dewil/occlu-mask/build/bin/warp-bicubic"' to
'WARP3="../collision_mask/build/bin/warp-bicubic"'

**Comment by Valery-Dewil** [comment]
Hi all, 
Yes, oooo1114 is right, you should modify the path where the collision mask repository was cloned (change "home/dewil/..." to yours. Basically, what oooo1114 proposed works: "../collision_mask/build/bin/warp-bicubic").
Regarding the imscript commands (plambda, downsa, blur, upsa), you must add the bin folder of imscript to your path: for instance you can add export PATH=$PATH:/path_to_imscript_folder/bin in you .bashrc file after having modified the path to the imscript folder.

**Comment by pankajsharma11** [comment]
Anyone getting error related to parallel -- bar in compute_mask.sh and tv1_flow.sh?
"parallel: invalid option -- '-'
parallel [OPTIONS] command -- arguments
	for each argument, run command with argument, in parallel
parallel [OPTIONS] -- commands
	run specified commands in parallel".
Thank you

**Comment by gfacciol** [comment]
It looks like the parallel command is not the same. 
if you run>   parallel --version    
you should get: 

> GNU parallel 20161222
> Copyright (C) 2007,2008,2009,2010,2011,2012,2013,2014,2015,2016
> Ole Tange and Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> GNU parallel comes with no warranty.
> 
> Web site: http://www.gnu.org/software/parallel
> 
> When using programs that use GNU Parallel to process data for publication
> please cite as described in 'parallel --citation'.

if you don't try installing it with 
> sudo apt-get install parallel

**Comment by ShaofengZou** [comment]
Thanks for your reply.
I wonder could you provide the imscript of plambda, downsa, blur, upsa in this repo? That's would help a lot.

**Comment by gfacciol** [comment]
these programs are in this repository:
https://github.com/mnhrdt/imscript

**Comment by ShaofengZou** [comment]
Thanks, got it. I found it in your ReadMe. Sorry.



zousf19@mails.tsinghua.edu.cn
 
From: gfacciol
Date: 2021-03-10 20:53
To: cmla/mf2f
CC: ShaofengZou; Author
Subject: Re: [cmla/mf2f] About warping result mask (#3)
these programs are in this repository:
https://github.com/mnhrdt/imscript

**Comment by ShaofengZou** [comment]
Another little question, have you try meta-learning method such as MAML on it? Recently I am working on that trying to speed up your algorithm.

Looking Forward to your reply. 



zousf19@mails.tsinghua.edu.cn
 
From: gfacciol
Date: 2021-03-10 20:08
To: cmla/mf2f
CC: ShaofengZou; Author
Subject: Re: [cmla/mf2f] About warping result mask (#3)
It looks like the parallel command is not the same.
if you run> parallel --version
you should get:
GNU parallel 20161222
Copyright (C) 2007,2008,2009,2010,2011,2012,2013,2014,2015,2016
Ole Tange and Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html
This is free software: you are free to change and redistribute it.
GNU parallel comes with no warranty.
Web site: http://www.gnu.org/software/parallel
When using programs that use GNU Parallel to process data for publication
please cite as described in 'parallel --citation'.
if you don't try installing it with
sudo apt-get install parallel

**Comment by pankajsharma47** [comment]
Thank you sir for your response, but still this parallel package in not installing neither in google Collab nor in windows.
"  File "/content/parallel-0.2.5/setup.py", line 5, in <module>
    import pprocess
  File "/content/parallel-0.2.5/pprocess.py", line 255
    raise AcknowledgementError, obj"

If anyone can help, it will be helpful.
Thank you all.

**Comment by pariasm** [comment]
In reply to ShaofengZou: We did try MAML and REPTILE to pre-train the network before the fine-tuning. We tried it on the single-frame case ([see this previous F2F paper](https://arxiv.org/abs/1811.12766)). The rationale was to have a better pre-trained network so as to reduce the number of fine-tuning iterations. For the meta-learning we considered denoising of three noise distributions (e.g. Gaussian, Poisson, correlated noise) with random parameters. It didn't work however. The meta-learned network required as many iterations as other pre-trained networks (e.g. a network pre-trained for Gaussian noise).

**Comment by pariasm** [comment]
In reply to pankajsharma11 and pankajsharma47

We tested this code in Linux (Ubuntu). The bash scripts (.sh files) won't run in Windows (although probably there are linux terminal emulators you could use, but I can't help you with that). 

However, if your problem is with parallel, you can just remove it. Substitute the loop in the `tvl1flow/tvl1flow.sh` script as follows:
```bash
for i in `seq $F $L`;
do
    ./tvl1flow `printf $I $((i))` \
        `printf $I $((i+1))` \
        `printf $O $((i))` \
        1 0.25 0.15 0.3 100 0 0.5 5 0.01 0; 
done
```
This will take longer as it will use a single core, but it will work. Hope this helps.

**Comment by ShaofengZou** [comment]
All right, thanks for your sharing. Looking forward a good method to speed it up !



zousf19@mails.tsinghua.edu.cn
 
From: Pablo Arias
Date: 2021-03-10 22:00
To: cmla/mf2f
CC: ShaofengZou; Author
Subject: Re: [cmla/mf2f] About warping result mask (#3)
In reply to ShaofengZou: We did try MAML and REPTILE to pre-train the network before the fine-tuning. We tried it on the single-frame case (see this previous F2F paper). The rationale was to have a better pre-trained network so as to reduce the number of fine-tuning iterations. For the meta-learning we considered denoising of three noise distributions (e.g. Gaussian, Poisson, correlated noise) with random parameters. It didn't work however. The meta-learned network required as many iterations as other pre-trained networks (e.g. a network pre-trained for Gaussian noise).

**Comment by pankajsharma47** [comment]
> In reply to pankajsharma11 and pankajsharma47
> 
> We tested this code in Linux (Ubuntu). The bash scripts (.sh files) won't run in Windows (although probably there are linux terminal emulators you could use, but I can't help you with that).
> 
> However, if your problem is with parallel, you can just remove it. Substitute the loop in the `tvl1flow/tvl1flow.sh` script as follows:
> 
> ```shell
> for i in `seq $F $L`;
> do
>     ./tvl1flow `printf $I $((i))` \
>         `printf $I $((i+1))` \
>         `printf $O $((i))` \
>         1 0.25 0.15 0.3 100 0 0.5 5 0.01 0; 
> done
> ```
> 
> This will take longer as it will use a single core, but it will work. Hope this helps.

Thank you for answer but it didn't work for me. Trying this in ubuntu now. 
Thank you sir.

**Comment by pankajsharma11** [comment]
Anyone can please let me how to install iio in ubantu? "Pip3 install iio" gives error. Any suggestions will be a great help.

**Comment by gfacciol** [comment]
You need to install some system dependencies first (compiler and
libraries):

sudo apt-get install build-essential libjpeg-dev libtiff-dev libpng-dev

**Comment by pankajsharma47** [comment]
Thank you sir.....iio is installed now. Set the path variable to imscript/bin still it shows no such directory.

**Comment by gfacciol** [comment]
You must download imscript from https://github.com/mnhrdt/imscript
and compile it calling the command make, that will create the directory
imscript/bin
